webapp/channels/src/components/admin_console/server_logs/logs.tsx

```diff
@@ -92,6 +94,23 @@ export default class Logs extends React.PureComponent<Props, State> {
         this.setState({page: this.state.page - 1});
     };
 
+    downloadLogs = async () => {
+        let logString;
+        if (this.props.isPlainLogs) {
+            await this.props.actions.getAllPlainLogs();
+            logString = this.props.allPlainLogs.join('');
+        } else {
```

Comment:
When testing the feature, I noticed that the download when using the plain view as a leading empty line. Can that get removed?

Reply:
Hm, let me re-test tomorrow

Reply:
The first image is from the plain logs.

![Screenshot from 2024-03-19 08-57-42](https://github.com/mattermost/mattermost/assets/16541325/246100e8-8a7e-474d-b12a-e45b892ea5ec)

Second image is from the JSON logs
![Screenshot from 2024-03-19 08-58-19](https://github.com/mattermost/mattermost/assets/16541325/dc49e884-8444-48f3-b845-b1d1b25083f1)

Please also note the different orderings in the first image.

Both issues (different ordering, leading empty line) are not a huge deal. But it would be nice to have them fixed.